13_sandboxes/test_case_generator.py

```diff
@@ -0,0 +1,380 @@
+# ---
+# cmd: ["modal", "run", "-m", "13_sandboxes.test_case_generator"]
+# args: ["--gh-owner", "modal-labs", "--gh-repo-name", "password-analyzer", "--gh-module-path", "src/password_strength", "--gh-tests-path", "tests", "--gh-branch", "main"]
+# ---
+import subprocess
+import time
+
+import modal
+
+app = modal.App(
+    name="sandbox-test-case-generator",
+)
+model_volume = modal.Volume.from_name("deepseek-model-volume", create_if_missing=True)
+files_volume = modal.Volume.from_name("files-volume", create_if_missing=True)
+
+MODEL_NAME = "deepseek-ai/deepseek-coder-6.7b-instruct"
+MODEL_REVISION = "e5d64addd26a6a1db0f9b863abf6ee3141936807"
+
+
+model_image = (
+    modal.Image.from_registry("lmsysorg/sglang:v0.4.9.post3-cu126", add_python="3.12")
+    .uv_pip_install(
+        "sglang[all]==0.4.9.post3",
+        "transformers==4.53.2",
+        "torch==2.7.1",
+        "accelerate==1.8.1",
+        "hf_transfer==0.1.9",
+        "numpy<2",
+    )
+    .env(
+        {
+            "HF_HUB_ENABLE_HF_TRANSFER": "1",
+            "HF_HOME": "/cache",
+        }
+    )
+    .entrypoint([])  # silence noisy logs
+)
+
+
+@app.cls(
+    image=model_image,
+    volumes={
+        "/cache": model_volume,
+        "/data": files_volume,
+    },
+    gpu="L40S",
+)
+@modal.concurrent(max_inputs=3)  # Each container runs up to 3 requests at once.
+class TestCaseServer:
+    @modal.enter()
+    def download_model(self):
+        from huggingface_hub import snapshot_download
+
+        snapshot_download(
+            MODEL_NAME,
+            local_dir=f"/cache/{MODEL_NAME}",
```

Comment:
Does `local_dir` work like `cache_dir`?

Reply:
yup! https://huggingface.co/docs/huggingface_hub/en/guides/download#download-files-to-a-local-folder